### ISSUE TITLE
Manage multiple campaigns/parties

### ISSUE THREAD

**Issue by mentonin** [Lurkars/gloomhavensecretariat#371]
- [x] Switching between campaigns (single-party)
- [x] Single campaign export/import
- [ ] Switching parties within same campaign
- [ ] Single party export/import
- [x] Switching characters within same party (implemented with absent characters)
- [x] Single character export/import

**Comment by Lurkars** [comment]
Here I have to admit that the current feature of party switching needs a re-labeling to campaign switching, because actually no data is shared between parties (as per rules, like global achievements for example), so actually this is campaign switching (and as labeled with party already available in the campaign management menu). I think that came from my initial misunderstanding of the rules (this feature is really old in comparison to some stuff). So actually, this issue should be more about: implement real party switching by keeping all global stuff etc (which is currently not the case). So happy for you: this feature is already implement, but in the bad side is labeled wrong and the expected feature from label is actually missing 😅

So to sum up the feature: a party is actually a single party campaign with separate stored
- achievements of any kind
- edition (so create a new party change edition from GH to FH and you have your GH campaign and FH campaign parallel in a single game state, and single game code with server feature usage)
- unlocks (scenarios, items, treasures etc) except characters which is wrong, just noticing
- battle goals config
- reputation, prosperity, buildings etc

So yeah, this needs rework of course and actually add real support for multiple parties in a single campaign and as first change, update all labels and texts about campaign and not party switching (and fix to move character unlocks also to the campaign and not global game state).

**Comment by Lurkars** [comment]
So as first step, in [v0.74.0](https://github.com/Lurkars/gloomhavensecretariat/releases/tag/v0.74.0) I renamed all the current functionality to what it is, a multi-campaign management. Next step is to do the real multi-party management on a campaign.

**Comment by mentonin** [comment]
That is nice. I think it would be wise to track what needs to be implemented and develop some tests. Off the top of my head, we are looking for the following features:

- [X] Easy switch between campaigns
- [ ] Easy switch between parties in a campaign
- [ ] Export subset of tracked campaigns
- [ ] Export subset of tracked parties
- [ ] Import parties into a campaign (without overwriting current parties)
- [ ] Import campaigns (without overwriting current campaigns)
- [ ] Saved on campaign level: scenario status, global achievements, treasures, buildings, available items, calendar status, inspiration, morale, town resources, unlocked characters, ...
- [ ] Saved on party level: character data, party achievements, used items, reputation, ...
- [ ] Saved on global level: ?

Honestly, I believe switching campaigns is much more useful than switching parties in a campaign, but that could be just me. Maybe see if someone needs the multi party functionality before working on it?

**Comment by Lurkars** [comment]
I checked the one point that should already be implemented like this. For the import export you're right, the import should not overwrite. So I will definitely change that (also need to switch label there too, it's import/export campaign currently not party). And you're also right (and that also caused the misunderstanding on my side), when people ask for multi party support they almost always mean multi campaign support and that why I implemented feature as it is. So I will also definitely not focus on real multi party support unless requested.

**Comment by Lurkars** [comment]
Okay, I think I'll close this. If multiple parties is requested someday, a new issue will be created.
